src/net/rpc.rs

```diff
@@ -420,7 +448,7 @@ impl RPCPoxInfoData {
         let cur_cycle_pox_active = sortdb.is_pox_active(burnchain, &burnchain_tip)?;
 
         Ok(RPCPoxInfoData {
-            contract_id: boot_code_id("pox", chainstate.mainnet).to_string(),
+            contract_id: boot_code_id(next_cycle_pox_contract, chainstate.mainnet).to_string(),
```

Comment:
Do you want to return `cur_cycle_pox_contract` or `next_cycle_pox_contract` here?

Reply:
Good catch, it should probably `cur_cycle_pox_contract`.

Although, clients will likely use the new `contract_versions` payload to determine Stacking tx construction, since blindly using the `contract_id` here can result in errors according to the PoX-2 phase transition plan.

Reply:
Fixed in [9fd5c77](https://github.com/stacks-network/stacks-blockchain/pull/3278/commits/9fd5c772c020f914e6e6cbd03ea4b62f1b01bc2a)